lib/active_merchant/billing/credit_card_methods.rb

```diff
@@ -39,10 +38,29 @@ module CreditCardMethods
         (491730..491759),
       ]
 
+      # https://www.mastercard.us/content/dam/mccom/global/documents/mastercard-rules.pdf, page 73
+      MASTERCARD_RANGES = [
+        (222100..272099),
+        (510000..559999),
+      ]
+
+      # https://www.mastercard.us/content/dam/mccom/global/documents/mastercard-rules.pdf, page 73
+      MAESTRO_RANGES = [
+        (639000..639099),
+        (670000..679999),
+      ]
+
       def self.included(base)
         base.extend(ClassMethods)
       end
 
+      def self.in_bin_range?(number, ranges)
+        bin = number.to_i
+        ranges.any? do |range|
+          range.include?(bin)
```

Comment:
We'll probably want to use `cover` instead of `include`, it's [faster](https://github.com/JuanitoFatas/fast-ruby/#range) since it doesn't traverse the entire range 🙂

Reply:
Oh, point. Will do!